packages/mui-base/src/Slider/SliderThumb/SliderThumb.tsx

```diff
@@ -0,0 +1,185 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { defaultRenderFunctions } from '../../utils/defaultRenderFunctions';
+import { getStyleHookProps } from '../../utils/getStyleHookProps';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import { resolveClassName } from '../../utils/resolveClassName';
+import { useForkRef } from '../../utils/useForkRef';
+import { useSliderContext } from '../Root/SliderProvider';
+import { SliderThumbProps } from './SliderThumb.types';
+import { useSliderThumb } from './useSliderThumb';
+
+const SliderThumb = React.forwardRef(function SliderThumb(
+  props: SliderThumbProps,
+  forwardedRef: React.ForwardedRef<HTMLSpanElement>,
+) {
+  const {
+    render: renderProp,
+    'aria-label': ariaLabel,
+    'aria-valuetext': ariaValuetext,
+    className,
+    disabled: disabledProp = false,
+    getAriaLabel,
+    getAriaValueText,
+    id,
+    ...otherProps
+  } = props;
+
+  const render = renderProp ?? defaultRenderFunctions.span;
+
+  const {
+    active: activeIndex,
+    'aria-labelledby': ariaLabelledby,
+    axis,
+    changeValue,
+    direction,
+    disabled: contextDisabled,
+    largeStep,
+    max,
+    min,
+    minStepsBetweenValues,
+    name,
+    orientation,
+    ownerState,
+    percentageValues,
+    scale,
+    step,
+    tabIndex,
+    values,
+  } = useSliderContext();
+
+  const mergedRef = useForkRef(typeof render === 'function' ? null : render.ref, forwardedRef);
```

Comment:
We have a hook for this: `useRenderPropForkRef`

Reply:
@michaldudak I did it manually here because `render` contains an additional `inputProps` that TS will complain about if I use `useRenderPropForkRef`, seemed better than `@ts-ignore`!

Reply:
Right! All good, then!